lib/index.js

```diff
@@ -20,19 +23,25 @@ module.exports = function (arg1, arg2) {
 
       generateOptions.sourceMap = createSourceMap
 
-      rollup(rollupOptions).then((bundle) => {
-        const result = bundle.generate(generateOptions)
-
-        file.contents = Buffer.from(result.code)
-
-        if (createSourceMap) {
-          result.map.file = entry
-          result.map.sources = result.map.sources.map(source => path.relative(file.cwd, source))
-          applySourceMap(file, result.map)
-        }
-
-        cb(null, file)
-      })
+      rollup(rollupOptions)
+        .then(bundle => {
+          const result = bundle.generate(generateOptions)
+
+          file.contents = Buffer.from(result.code)
+
+          if (createSourceMap) {
+            result.map.file = entry
+            result.map.sources = result.map.sources.map(source => path.relative(file.cwd, source))
+            applySourceMap(file, result.map)
+          }
+
+          cb(null, file)
+        }).catch(err => {
+          process.nextTick(() => {
+            this.emit('error', new PluginError(PLUGIN_NAME, err))
+            cb(null, file)
```

Comment:
Oh, it is awesome code!
I understand. I'm going to use it.
https://github.com/gulpjs/gulp/tree/master/docs/writing-a-plugin#what-it-does